lib/acts_as_taggable_on/taggable/core.rb

```diff
@@ -107,8 +107,12 @@ def grouped_column_names_for(object)
       self.class.grouped_column_names_for(object)
     end
 
+    def unpersisted_custom_contexts
+      @unsaved_custom_contexts ||= []
+    end
+
     def custom_contexts
-      @custom_contexts ||= taggings.map(&:context).uniq
+      @custom_contexts ||= (taggings.map(&:context) + unpersisted_custom_contexts).uniq
```

Comment:
Uhm, if you use `pluck`, then the taggings model won't be cached. So, accessing to it again will result in another query.

Not sure if it's better or worse :(